app/utils/seda_signer.go

```diff
@@ -0,0 +1,133 @@
+package utils
+
+import (
+	"bytes"
+	"fmt"
+	"sort"
+
+	ethcrypto "github.com/ethereum/go-ethereum/crypto"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	pubkeytypes "github.com/sedaprotocol/seda-chain/x/pubkey/types"
+)
+
+type SEDASigner interface {
+	GetValAddress() sdk.ValAddress
+	Sign(input []byte, index SEDAKeyIndex) (signature []byte, err error)
+	ReloadIfMismatch(pubKeys []pubkeytypes.IndexedPubKey) error
+	IsLoaded() bool
+}
+
+var _ SEDASigner = &sedaKeys{}
+
+type sedaKeys struct {
+	valAddr  sdk.ValAddress
+	keys     map[SEDAKeyIndex]indexedPrivKey
+	pubKeys  []pubkeytypes.IndexedPubKey // sorted by index
+	keyPath  string
+	isLoaded bool
+}
+
+// LoadSEDASigner loads the SEDA keys from a given file path and
+// returns a SEDASigner interface.
+func LoadSEDASigner(keyFilePath string) (SEDASigner, error) {
+	keys, err := loadSEDAKeys(keyFilePath)
+	if err != nil {
+		keys.keyPath = keyFilePath
```

Comment:
Why do we do this here again? Should be the same as what is set on line 66 right?

Reply:
Not if `loadSEDAKeys()` results in an error though.